src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -285,3 +286,40 @@ GetCitusDependedObjectArgs(int pgMetaTableVarno, int pgMetaTableOid)
 
 	return list_make2((Node *) metaTableOidConst, (Node *) oidVar);
 }
+
+
+/*
+ * ShouldExecutePropagationSteps decides if we should execute preprocess,
+ * postprocess and qualify steps.
+ *
+ * If EnableUnsupportedFeatureMessages is set true, it returns true.
+ * Otherwise, we use the object's address method to see if all the addresses
+ * in the returned list are valid. If all addresses are valid, the method
+ * returns true; else, false.
+ */
+bool
+ShouldExecutePropagationSteps(Node *node, const DistributeObjectOps *ops)
+{
+	if (EnableUnsupportedFeatureMessages)
```

Comment:
Do we really need this? I think we should be fine (and good) to have this function run always?

Reply:
All other tests whose output depend on preprocess step fails if I remove the flag. That is why we need the flag to prevent preprocess in only while executing pg vanilla tests.